sources/oracle/toddl.go

```diff
@@ -88,12 +88,21 @@ func toSpannerTypeInternal(conv *internal.Conv, spType string, srcType string, m
 		case ddl.String:
 			return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, nil
 		default:
-			// If no scale is avalible then map it to int64, and numeric elsewhere.
-			if len(mods) == 1 && mods[0] >= 1 && mods[0] < 19 {
-				return ddl.Type{Name: ddl.Int64}, nil
-			} else {
+			modsLen := len(mods)
+			if modsLen == 0 {
 				return ddl.Type{Name: ddl.Numeric}, nil
+			} else if modsLen == 1 { // Only precision is available.
+				if mods[0] > 29 {
+					// Max precision in Oracle is 38. String representation of the number should not have more than 50 characters
+					return ddl.Type{Name: ddl.String, Len: 50}, nil
+				}
+				return ddl.Type{Name: ddl.Int64}, nil
+			} else if mods[0] > 29 || mods[1] > 9 { // When both precision and scale are available and within limit
+				// Max precision in Oracle is 38. String representation of the number should not have more than 50 characters
+				return ddl.Type{Name: ddl.String, Len: 50}, nil
 			}
```

Comment:
Do we need an error in else to handle scenarios where none of the 3 conditions in this block are satisfied ?

Reply:
If none of the condition satisfies then it should be Numeric. That is if both precision and scale available and precision <=29 and scale <=9.